From cd7baa971adc0387afc9187c4f0c5aa092129fa3 Mon Sep 17 00:00:00 2001
From: Heinrich Schuchardt <xypron.glpk@gmx.de>
Date: Thu, 6 Jun 2019 01:00:40 +0200
Subject: [PATCH 1/1] efi_loader: wait for key may not signal itself

efi_key_notify() is the notification function of the wait for key event. It
may not call any function that in turn will signal the wait for key event.
Otherwise with a proper event queue we may end up in an endless loop.

Remove the body of the function.

Signed-off-by: Heinrich Schuchardt <xypron.glpk@gmx.de>
---
 lib/efi_loader/efi_console.c | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/lib/efi_loader/efi_console.c b/lib/efi_loader/efi_console.c
index 3b7578f3aa..73b291d535 100644
--- a/lib/efi_loader/efi_console.c
+++ b/lib/efi_loader/efi_console.c
@@ -1067,9 +1067,6 @@ static void EFIAPI efi_console_timer_notify(struct efi_event *event,
  */
 static void EFIAPI efi_key_notify(struct efi_event *event, void *context)
 {
-	EFI_ENTRY("%p, %p", event, context);
-	efi_cin_check();
-	EFI_EXIT(EFI_SUCCESS);
 }
 
 /**
-- 
2.20.1

